units/Unit.gd

```diff
@@ -87,6 +98,9 @@ func restore_current_moves():
 	current_moves = base_max_moves
 	can_attack = true
 
+func has_moved():
```

Comment:
You ask "has the unit moved" so it should be `return current_moves < base_max_moves`